lib/providers/storage.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:concordium_wallet/entities/accepted_terms_and_conditions.dart';
+import 'package:concordium_wallet/state/network.dart';
+import 'package:hive_flutter/hive_flutter.dart';
+
+/// Service for interacting with [Hive].
+class StorageProvider {
+  final Box<AcceptedTermsAndConditions> _acceptedTermsAndConditionBox;
+
+  const StorageProvider._(this._acceptedTermsAndConditionBox);
+
+  static Future<StorageProvider> init() async {
+    await Hive.initFlutter();
+    _registerAdapters();
+    await _openBoxes();
+
+    return StorageProvider._(await Hive.openBox<AcceptedTermsAndConditions>(AcceptedTermsAndConditions.table));
```

Comment:
It might be because I don't know Hive well (yet), but you also open the box in `_openBoxes()`. So to me it appears like we do that twice which is weird.

Reply:
So I just checked the documentation and it states that:
```
If the box is already open, the instance is returned and all provided parameters are being ignored.
```
But for clarity we should use:
```
    return StorageProvider._(Hive.box<AcceptedTermsAndConditions>(AcceptedTermsAndConditions.table));
```
instead.

Reply:
The idea was to prepare for more boxes to be open asynchronously in `_openBoxes`. I didn’t want to change to return type every time a new box was added.

I changed to your fix by changing the input to the constructor.